File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/RankTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.ranks
import kotlin.test.Test
import kotlin.test.assertEquals

class RankTest {
	@Test
	fun `Ranks array should be valid`() {
		for (rank in ranks.indices) {
			assertEquals(ranks[rank], rank)
		}
	}
}

File: pulse-go/internal/pulse/rank_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestRank(t *testing.T) {
	t.Run("Ranks array should be valid", func(t *testing.T) {
		for rank := range Ranks {
			if Ranks[rank] != rank {
				t.Errorf("wanted ranks at index %d to be %v, got %v", rank, rank, Ranks[rank])
			}
		}
	})
}
